# Match method names to the decorated API routes (`nestjs-pedantic/match-methods-to-routes`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŒ `all`, âœ… `recommended`.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## Rule Details

âŒ **Incorrect** code:

```ts
@Controller("user")
class UserController {
  @Get("route")
  someRandomName() {}

  @Post("other-route/:userId")
  someOtherRandomName() {}
}
```

âœ… **Corrrect** code:

```ts
@Controller("user")
class UserController {
  @Get("route")
  getRoute() {}

  @Post("other-route/:userId")
  postOtherRouteByUserId() {}
}
```
