# Disallow duplicate route parameters (`nestjs-pedantic/no-duplicate-route-params`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŒ `all`, âœ… `recommended`.

<!-- end auto-generated rule header -->

## Rule Details

âŒ **Incorrect** code:

```ts
@Controller("user")
class UserController {
  @Get(":id/route/:id")
  getByIdRouteById() {}
}
```

âœ… **Corrrect** code:

```ts
@Controller("user")
class UserController {
  @Get("route/:id")
  getRouteById() {}
}
```
