# Disallow unused route parameters (`nestjs-pedantic/no-unused-route-params`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŒ `all`, âœ… `recommended`, âš¡ `recommendedSwc`.

ğŸ’¡ This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).

<!-- end auto-generated rule header -->

## Rule Details

âŒ **Incorrect** code:

```ts
@Controller("user")
class UserController {
  @Get(":id")
  getById() {}

  @Put("details/:id")
  putDetailsById() {}

  @Post("other/:userId")
  postOtherByUserId() {}
}
```

âœ… **Corrrect** code:

```ts
@Controller("user")
class UserController {
  @Get(":id")
  getById(@Param("id") id: string) {}

  @Put("details")
  putDetails() {}

  @Post("other/:userId")
  postOtherByUserId(@Param() params: Record<string, string>) {}
}
```
